### PR TITLE
nxcomp: Restore old 'VERSION' move behavior

### PR DESCRIPTION
**Pull request by arjanvandervelde**
#### Description

This restores old behavior where VERSION file was move at "pre-build" time, rather than "post-patch". VERSION is used to fill in version number. It's currently broken and version ends up being "0.0.0.0".

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.7.2 23H311 arm64

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ryandesign** [comment]
> * [x]  followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?

The guidelines have not been followed. The commit message should begin with the port name, a colon, and a space.

If this changes the files the port installs, the `revision` must be increased.

I'm surprised the build goes wrong when VERSION is moved before configure time. If it was reading the VERSION file at configure time, I would have expected an error message when it is not found. But indeed I do see `sh: VERSION: No such file or directory` while autoreconf is running, and unfortunately it does not cause the build to exit with an error code. Maybe we can fix it so that it does exit with an error when something goes wrong.

I changed it from `pre-build` to `post-patch` for a reason: to stop an unexpected error if the build is interrupted and then restarted. This PR will reintroduce that problem. Instead of `pre-build`, `post-configure` would be a better place that would not reintroduce the problem.

Or we can rename the VERSION file to something else, like VERSION.txt.

The problem (the incompatibility with C++20 compilers that necessitates renaming this file, or removing the `-I` flag pointing to the directory it's in) needs to be reported to the developers so they can fix it properly so we can remove our workarounds.

**Comment by ryandesign** [comment]
> The […] incompatibility with C++20 compilers […] needs to be reported to the developers

https://github.com/ArcticaProject/nx-libs/issues/1086

**Comment by agvandervelde** [comment]
Thanks for looking into this. I've updated to commit message, apologies for that. I think `post-configure` sounds reasonable. I'll update that and the revision, and test the build.

**Comment by arjanvandervelde** [comment]
Thanks for looking into this. I've updated to commit message, apologies for that. I think post-configure sounds reasonable. I'll update that and the revision, and test the build.

**Comment by arjanvandervelde** [comment]
@ryandesign, done. thank you.

**Comment by agvandervelde** [comment]
done.

**Comment by arjanvandervelde** [comment]
done.

**Review by ryandesign — APPROVED** [review]

**Comment by arjanvandervelde** [comment]
@ryandesign would you be able to merge this, or is this awaiting maintainer approval? thanks!

**Comment by ryandesign** [comment]
I was awaiting maintainer approval but since I broke it without their approval I suppose I should fix it without their approval too.

**Pull request merged**
